File: Python/62. Unique Paths.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        if m == 0 or n == 0:
            return 0
        paths = [[0 for i in range(n)] for j in range(m)]
        paths[0][0] = 1
        for i in range(0, m):
            for j in range(0, n):
                paths[i][j] += 0 if i == 0 else paths[i - 1][j]
                paths[i][j] += 0 if j == 0 else paths[i][j - 1]
        return paths[m - 1][n - 1]

File: 2021/62. Unique Paths.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        if (m == 0 || n == 0) return 0;
        int dp[m][n];
        for (int i = 0; i <m; i++) dp[i][0] = 1;
        for (int j = 0; j <n; j++) dp[0][j] = 1;
        
        
        for (int i = 1; i < m; i++) {
            for (int j = 1; j < n; j++) {
                dp[i][j] = dp[i][j - 1] + dp[i - 1][j];
            }
        }
        return dp[m - 1][n - 1];
    }
};

class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<vector<int>> dp(m + 1, vector<int>(n + 1, 0));        
        for (int i = 0; i < m; i++) {            
            for (int j = 0; j < n; j++) {        
                if (i == 0 && j == 0) {
                    dp[i + 1][j + 1] = 1;
                } else {
                    dp[i + 1][j + 1] = dp[i][j + 1] + dp[i + 1][j];
                }                
            }
        }
        return dp[m][n];
    }
};